traffic_ops/traffic_ops_golang/server/servers_server_capability.go

```diff
@@ -0,0 +1,158 @@
+package server
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/lib/go-tc/tovalidate"
+	"github.com/apache/trafficcontrol/lib/go-util"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/dbhelpers"
+	validation "github.com/go-ozzo/ozzo-validation"
+)
+
+const (
+	ServerCapabilityQueryParam = "serverCapability"
+	ServerQueryParam           = "serverId"
+	ServerHostNameQueryParam   = "serverHostName"
+)
+
+// we need a type alias to define functions on
+type TOServerServerCapability struct {
+	api.APIInfoImpl `json:"-"`
+	tc.ServerServerCapability
+}
+
+func (ssc *TOServerServerCapability) SetLastUpdated(t tc.TimeNoMod) { ssc.LastUpdated = &t }
+func (ssc *TOServerServerCapability) NewReadObj() interface{} {
+	return &tc.ServerServerCapability{}
+}
+func (ssc *TOServerServerCapability) SelectQuery() string { return scSelectQuery() }
+func (ssc *TOServerServerCapability) ParamColumns() map[string]dbhelpers.WhereColumnInfo {
+	return map[string]dbhelpers.WhereColumnInfo{
+		ServerCapabilityQueryParam: dbhelpers.WhereColumnInfo{"sc.server_capability", nil},
+		ServerQueryParam:           dbhelpers.WhereColumnInfo{"s.id", api.IsInt},
+		ServerHostNameQueryParam:   dbhelpers.WhereColumnInfo{"s.host_name", nil},
+	}
+
+}
+func (ssc *TOServerServerCapability) DeleteQuery() string { return scDeleteQuery() }
+
+func (ssc TOServerServerCapability) GetKeyFieldsInfo() []api.KeyFieldInfo {
+	return []api.KeyFieldInfo{
+		{ServerQueryParam, api.GetIntKey},
+		{ServerCapabilityQueryParam, api.GetStringKey},
+	}
+}
+
+// Need to satisfy Identifier interface but is a no-op as path does not have Update
+func (ssc TOServerServerCapability) GetKeys() (map[string]interface{}, bool) {
```

Comment:
I think this is used when creating changelog messages by the shared handlers. You could probably mimic what the deliveryservice servers handler does (traffic_ops_golang/deliveryservice/servers/servers.go GetKeys on L60)

Reply:
ahh your right modification coming up